src/ansys/aedt/core/visualization/plot/pyvista.py

```diff
@@ -985,6 +1000,135 @@ def add_field_from_data(
         filedata.point_data[self.fields[-1].label] = np.array(fields_data)
         self.fields[-1]._cached_polydata = filedata
 
+    @pyaedt_function_handler()
+    def _read_case(self, field):
+        reader = pv.get_reader(os.path.abspath(field.path)).read()
+        field._cached_polydata = reader[reader.keys()[0]].extract_surface()
+
+        if (
+            hasattr(field._cached_polydata.point_data, "active_vectors")
+            and field._cached_polydata.point_data.active_vectors_name
+        ):
+            field.scalar_name = field._cached_polydata.point_data.active_vectors_name
+            field.vector_scale = (max(field._cached_polydata.bounds) - min(field._cached_polydata.bounds)) / (
+                50
+                * (
+                    np.vstack(field._cached_polydata.active_vectors).max()
+                    - np.vstack(field._cached_polydata.active_vectors).min()
+                )
+            )
+            field._cached_polydata["vectors"] = field._cached_polydata.active_vectors
+
+            field.is_vector = True
+        else:
+            field.scalar_name = field._cached_polydata.point_data.active_scalars_name
+
+    @pyaedt_function_handler()
+    def _read_aedtplt(self, field):
+        vertices, faces, scalars, log1 = _parse_aedtplt(field.path)
+        if self.convert_fields_in_db:
+            scalars = [np.multiply(np.log10(i), self.log_multiplier) for i in scalars]
+        fields_vals = pv.PolyData(vertices[0], faces[0])
+        field._cached_polydata = fields_vals
+        if isinstance(scalars[0], list):
+            field.vector_scale = (max(fields_vals.bounds) - min(fields_vals.bounds)) / (
+                50 * (np.vstack(scalars[0]).max() - np.vstack(scalars[0]).min())
+            )
+            field._cached_polydata["vectors"] = np.vstack(scalars).T
+            field.label = "Vector " + field.label
+            field._cached_polydata.point_data[field.label] = np.array(
+                [np.linalg.norm(x) for x in np.vstack(scalars[0]).T]
+            )
+            try:
+                field.scalar_name = field._cached_polydata.point_data.active_scalars_name
+                field.is_vector = True
+            except Exception:
+                field.is_vector = False
+        else:
+            field._cached_polydata.point_data[field.label] = scalars[0]
+            field.scalar_name = field._cached_polydata.point_data.active_scalars_name
+            field.is_vector = False
+        field.log = log1
+
+    @pyaedt_function_handler()
+    def _read_fld(self, field):
+        nodes = []
+        values = []
+        is_vector = False
+        with open_file(field.path, "r") as f:
+            try:
+                lines = f.read().splitlines()[field.header_lines :]
+                if ".csv" in field.path:
```

Comment:
```suggestion
                if field.path and field.path.endswith(".csv"):
```
A field can be added from data and, in that case, there are no path. See `add_field_from_data`